Question 6 and 8

Yicaa opened the pull request:
Question 3 and 8 done. Appreciate if anyone can help to check it ;)

parolidiego reviewed (approved):
Seems good to me! 

Let's wait for another review to double check it.

bm-go commented:
Looks good to me but I committed a change so can't approve/merge. The <45% votes are mostly due to one 2008 boycott which is quite interesting 👍  If someone else can review/approve, it's good :)

Yicaa commented:
Thank u both for the review! And especially thank u Brad for doing the research and the insight of 2008 boycott. I'm sorry for overlapping other groups' questions, but I will comment this to Linghan and Jia and recommend them to add this summary of context, which is very very interesting!

Yicaa commented:
Opening new pull request for question **6 and 8**.
Deleted previous overlapped question 3.
Updated some changes of written form suggested by Brad in Q8.

Yicaa commented:
Finished the Q6. Really appreciate it if someone can help review and approve!

bm-go commented:
Have reviewed the Q6 code as well and looks all good. Committed one change to refer to valid_codes instead of total_codes.. might have introduced a version control issue though Yijia which I will try fix now.. :)
